### ISSUE TITLE
Output missing assembled contigs

### ISSUE THREAD

**Issue by JFWolters** [ndierckx/NOVOPlasty#87]
Hello,

Thank you for your prior assistance, NOVOplastly3.2.pl fixed the majority of my assembly issues and thus I closed that issue.

I do have one odd assembly remaining that I think is due to different problems. For this assembly I have intermittently had issues where the job is either killed or gives an out of memory error, despite 64 - 256 GB RAM on the compute clusters I am using.

However, I was able to get it to run successfully once but it output only 1 very short contig. The log shows several longer contigs being assembled but these are not present in the output.

The input seed is a mitochondrial contig assembled with plasmidSPAdes which I have verified as containing all expected mitochondrial genes. It appears that NOVOplastly assembled contigs close in size to this seed but for some reason they are absent from the output.

Do you have any thought on what may be causing this behavior?

I have included the regular and extended log, and the output from STDOUT and STDERROR in run_log.txt which shows the assembly of the missing contigs.

The reads used as input are available in SRA as SRR6475985

[Citeromyces_hawaiiensis.zip](https://github.com/ndierckx/NOVOPlasty/files/3253922/Citeromyces_hawaiiensis.zip)

**Comment by ndierckx** [comment]
Hi, there is a small region with very low coverage, where it goes wrong. But you can find the large contig in the contig_tmp file, this file is a bit of a backup when not all contigs are being outputted. Although it is a bit shorter then your other assembly. Maybe try this seed to see if you can assemble the rest TAAGATCATCCTCTTTATTGGAATATAGTCCTATAATTTTATTATTAATTTTATGAATTG
GTGGTTTATCAACATTAATATCAGGTTTAATAGCAACAGTATCAAATGATATTAAAAAAA
TAATAGCTTTATCTACAATGAGTCAATTAGGTATAATGATGATAGCTATAGGTATATCAG
GTTATAATTTAGCATTATATCATTTAATTTGTCATAGTTTCTTTAAAGCTTTATTATTTA
TGTCAGCTGGTTCAATAATTCATAGTTTCTTATCTGAATATCAAGATATTAGAGTTTATG

**Comment by JFWolters** [comment]
Hi,

Thanks for the info about the temporary contigs, I can at least try to rescue the assembly from there.

I tried the alternative seed and got an error:
"Quantifier follows nothing in regex; marked by <-- HERE in m/* <-- HERE .G.....CAATAAGAATCAAA.TTCGAAAGCTCCTTCA/ at ../../NOVOPlasty3.2.pl line 6692."

I've attached the output with the extended log and STDOUT and STDERR in run_log.txt.

[Citeromyces_hawaiiensis_seed_test.zip](https://github.com/ndierckx/NOVOPlasty/files/3277643/Citeromyces_hawaiiensis_seed_test.zip)

**Comment by ndierckx** [comment]
That's a bug, Thanks for reporting! Will upload new version in a minute

**Comment by ndierckx** [comment]
I uploaded 3.3, not sure if the problem will be resolved. if it doesn't let me know

**Comment by JFWolters** [comment]
I gave 3.3 a try and it did prevent it from erroring out in the same way.

The assembly was still killed for reasons that aren't clear. The error log just says "Killed" again. The large contigs still appear to be there in the temporary contigs file.

Ive attached that run with the extended log.

STDOUT is in NOVOPlasty_test_606576_0.log
STDERR is in NOVOPlasty_test_606576_0.err
[Citeromyces_hawaiiensis_3.3_seed_test.zip](https://github.com/ndierckx/NOVOPlasty/files/3287488/Citeromyces_hawaiiensis_3.3_seed_test.zip)

**Comment by ndierckx** [comment]
I have no idea why it gets killed.. Only reason I can think is some bug that causes a loop and fills up the memory, but I don't see any memory peak in your output. How large are your files?

**Comment by JFWolters** [comment]
The reads files are 931 MB each with about 1.78 million read pairs.

Its strange that it appears to end while trying to assemble an additional large contig after already assembling one that I believe is the correct Mt sequence. The 2nd contig in tmp contigs file matches the mitochondrial contig from another assembler and even includes additional sequence that spans the gap missing from the other assembly. The contig also appears circular based on the overlap with the other assembly. Other than the crash it seems NOVOplasty is performing well.

I don't see the seed showing up in the nuclear assembly of this species so its probably not some weird case of the seed being part of a duplication in different genomes.

In any case this output is usable with a bit of manual adjustment. Let me know if you want me to close the issue since my initial question has been clearly answered and the crash is likely a separate issue.

**Comment by ndierckx** [comment]
is it the SRR6475985 dataset? I could try myself then

**Comment by JFWolters** [comment]
Yes, those are the reads for this assembly. Please give it a shot, any assistance would be greatly appreciated.

I learned a little more about the job scheduler I'm using and my jobs have likely been targeted first for removal if the cluster ran into memory limitations. The job appears to have been using 14889 MB when it was killed. I'm not sure if that is unreasonably high or within the expected range. I will give it another go with a high amount requested.

**Comment by ndierckx** [comment]
15 GB is not that much, although seems a bit much for this small dataset (only 322 MB per file, right?)
I will download it give a try later

**Comment by JFWolters** [comment]
I'm getting 1.1 GB (333 MB gzipped) for the reads from SRA. There is a size discrepancy between my local version of the reads and the SRA version (I'm working in the same lab as the publisher of this data) so I will double check on my end by running again using the SRA reads.

**Comment by JFWolters** [comment]
I ran the assembly again and received an error similar to the one I previously got with NOVOPlasty3.2.pl.
"Quantifier follows nothing in regex; marked by <-- HERE in m/* <-- HERE .G.....CAATAAGAATCAAA.TTCGAAAGCTCCTTCA/ at ../../NOVOPlasty3.3.pl line 6700."

The run took several days and consumed quite an incredible amount of memory, hitting 196.287 GB at the time it crashed.

The script is crashing while trying to construct a very large contig (>170 kb), when I believe its already successfully assembled the mtDNA at around 47kb.

I'm still getting everything I need from contigs_tmp file but I figured best to report anyway.

Ive attached all the output.
STDOUT is in NOVOPlasty_test_606712_0.out
STDERR is in NOVOPlasty_test_606712_0.err
A log of the job with information like memory usage is in NOVOPlast_test_606712.log

The extended log is in log_extended_Citeromyces_hawaiiensis.txt


[Citeromyces_hawaiiensis_3.3_seed_test_highMEM.zip](https://github.com/ndierckx/NOVOPlasty/files/3325980/Citeromyces_hawaiiensis_3.3_seed_test_highMEM.zip)

**Comment by ndierckx** [comment]
Hi, thanks for reporting!
I know what the problem is but i al currently on holiday so can't fix it at the moment. I found it just before I left, seems there are very high concentrations of ambiguous positions in this assembly and the program is trying to many options to resolve this. I will have to make a failsafe to limit this because just takes too long and requires too much memory for no improvement

**Comment by ndierckx** [comment]
Hi,
Got back from holiday and I think I solved the problem.
It should go much faster and use much less memory
You can download the updated version (3.4)

**Comment by JFWolters** [comment]
Hello,

I ran another test using 3.4 and it appears to be having similar memory issues. In this case it reached 170GB memory usage before being killed by the job manager.

The tmp contigs contains several very similar contigs that match expected mitochondrial sequence which I'm guessing is a result of the high concentration of ambiguous positions you mentioned. Heteroplasmy isn't known in yeast but these species haven't had their mtDNA closely studied so its a possibility.

I've attached all the output:
STDOUT is in NOVOPlasty_test_612311_0.out
STDERR is in NOVOPlasty_test_612311_0.err
A log of job information and memory usage is in NOVOPlasty_test_612311.log

The extended log is in log_extended_Citeromyces_hawaiiensis.txt

Let me know if there is any more information I can provide.

[Citeromyces_hawaiiensis_3.4_seed_test_highMEM.zip](https://github.com/ndierckx/NOVOPlasty/files/3383185/Citeromyces_hawaiiensis_3.4_seed_test_highMEM.zip)
